clarity/src/vm/functions/crypto.rs

```diff
@@ -62,8 +63,30 @@ macro_rules! native_hash_func {
     };
 }
 
+macro_rules! native_hash_func_new {
```

Comment:
This is consensus-critical code.  I'd strongly prefer if you did not add this, given that the existing code already works (and if it has bugs, it's better to be stuck with them until the next hard fork than risk a chain split).